bench/src/main/scala/cats/collections/bench/DietToIteratorRegressionBench.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.collections
+package bench
+
+import org.openjdk.jmh.annotations._
+import scala.util.Random
+
+@State(Scope.Benchmark)
+class DietToIteratorRegressionBench {
```

Comment:
Since the implementation of `toIterator` has changed tremendously, I would like to make sure that there is no performance downgrade. Thus, with help of this new benchmark I measured throughput of the `toIterator` method for the old vs new versions. The numbers are below (Scala 2.13.10, openjdk64-11.0.16.1):

Old `toIterator`:
```
Benchmark                                 (shiftBits)  (size)   Mode  Cnt        Score        Error  Units
DietToIteratorRegressionBench.toIterator            0     100  thrpt   25  9454277.327 ± 202820.369  ops/s
DietToIteratorRegressionBench.toIterator            0    1000  thrpt   25  9180334.422 ± 168053.101  ops/s
DietToIteratorRegressionBench.toIterator            0   10000  thrpt   25  9435667.955 ± 175937.003  ops/s
DietToIteratorRegressionBench.toIterator            8     100  thrpt   25    13275.529 ±    116.664  ops/s
DietToIteratorRegressionBench.toIterator            8    1000  thrpt   25     2316.657 ±     21.123  ops/s
DietToIteratorRegressionBench.toIterator            8   10000  thrpt   25  9253691.385 ± 125554.986  ops/s
DietToIteratorRegressionBench.toIterator           16     100  thrpt   25    11878.898 ±     52.812  ops/s
DietToIteratorRegressionBench.toIterator           16    1000  thrpt   25      776.785 ±      4.678  ops/s
DietToIteratorRegressionBench.toIterator           16   10000  thrpt   25       62.205 ±      0.463  ops/s
DietToIteratorRegressionBench.toIterator           24     100  thrpt   25    11872.484 ±     13.432  ops/s
DietToIteratorRegressionBench.toIterator           24    1000  thrpt   25      773.976 ±      8.582  ops/s
DietToIteratorRegressionBench.toIterator           24   10000  thrpt   25       59.295 ±      0.426  ops/s
```
New `toIterator`:
```
Benchmark                                 (shiftBits)  (size)   Mode  Cnt         Score        Error  Units
DietToIteratorRegressionBench.toIterator            0     100  thrpt   25  29247017.785 ± 238590.018  ops/s
DietToIteratorRegressionBench.toIterator            0    1000  thrpt   25  28888641.861 ± 390222.792  ops/s
DietToIteratorRegressionBench.toIterator            0   10000  thrpt   25  29404256.921 ±  82854.894  ops/s
DietToIteratorRegressionBench.toIterator            8     100  thrpt   25    262861.156 ±   1205.228  ops/s
DietToIteratorRegressionBench.toIterator            8    1000  thrpt   25     58579.311 ±   1457.804  ops/s
DietToIteratorRegressionBench.toIterator            8   10000  thrpt   25  28011756.184 ± 528586.551  ops/s
DietToIteratorRegressionBench.toIterator           16     100  thrpt   25    234386.167 ±   4620.642  ops/s
DietToIteratorRegressionBench.toIterator           16    1000  thrpt   25     21461.643 ±    220.749  ops/s
DietToIteratorRegressionBench.toIterator           16   10000  thrpt   25      2107.130 ±     59.657  ops/s
DietToIteratorRegressionBench.toIterator           24     100  thrpt   25    235759.491 ±   2618.151  ops/s
DietToIteratorRegressionBench.toIterator           24    1000  thrpt   25     21929.967 ±    430.028  ops/s
DietToIteratorRegressionBench.toIterator           24   10000  thrpt   25      2121.011 ±     36.403  ops/s
```
So I would say that the performance looks quite a bit improved in there.